lib/auth/users/usersv1/okta.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package usersv1
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+)
+
+// checkOktaUpdateAccess tests that, if the identity provided in authzCtx has
+// the built-in Okta role, that the "origin: Okta" label is present on both the
+// existing user resource and the new value it will take after any update
+// completes. If either of these records is missing the label, this function
+// returns an error.
+func (s *Service) checkOktaUpdateAccess(ctx context.Context, authzCtx *authz.Context, new types.User, existingUsername string) error {
```

Comment:
nit: I don't think we need `existingUsername` here, can just get from `new` (types.User)